CMakeLists.txt

```diff
@@ -0,0 +1,104 @@
+cmake_minimum_required(VERSION 3.10)
+
+set(CMAKE_OSX_DEPLOYMENT_TARGET 10.13)  # Has to be set before `project()`, and ignored on non-macos
+
+project(lokinet-gui
+    VERSION 0.2.2
+    DESCRIPTION "Lokinet graphical control panel"
+    HOMEPAGE_URL https://github.com/loki-project/loki-network-control-panel
+    LANGUAGES CXX)
+
+find_package(Qt5 COMPONENTS Qml Quick Widgets Charts REQUIRED)
+find_package(Threads)
+
+add_executable(lokinet-gui
+    src/main.cpp
+    src/QmlClipboardAdapter.cpp
+    src/HttpClient.cpp
+    src/LokinetApiClient.cpp
+    src/ApiPoller.cpp
+    src/PlatformDetails.cpp
+    src/BandwidthChartData.cpp
+    src/process/LokinetProcessManager.cpp
+
+    res/resources.qrc
+    )
+
+target_link_libraries(lokinet-gui PRIVATE
+    Qt5::Core Qt5::Qml Qt5::Quick Qt5::Widgets Qt5::Charts
+    Threads::Threads
+    )
+
+set_target_properties(lokinet-gui
+    PROPERTIES
+        CXX_STANDARD 14
+        CXX_EXTENSIONS OFF
+        CXX_STANDARD_REQUIRED ON
+
+        AUTORCC ON  # Enables processing *.qrc in sources
+        AUTOMOC ON  # Does... whatever automoc is
+    )
+
+if(APPLE)
+    option(MACOS_APP "Build a LokinetGUI.app bundle instead of a binary" OFF)
+    set(MACOS_SIGN ""
+        CACHE STRING "enable codesigning -- use a 'Apple Distribution' key (or key description) from `security find-identity -v`")
+
+    target_sources(lokinet-gui PRIVATE src/process/MacOSLokinetProcessManager.cpp)
+
+    if(MACOS_APP)
+        add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lokinet-gui.icns
+            COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/tools/mk-icns.sh ${CMAKE_CURRENT_SOURCE_DIR}/res/images/icon.svg ${CMAKE_CURRENT_BINARY_DIR}/lokinet-gui.icns
```

Comment:
Will this be invoked every time, or only when the `icns` file/dir/whatever needs to be generated?

Reply:
The latter.